src/dev_server/index.tsx

```diff
@@ -53,40 +53,53 @@ const DevServer = props => {
   const proxyHandler = (socket) => {
     return {
       get: (target, property) => {
-        if (property in target && target[property] !== null) {
-          const value = target[property];
-
-          if (typeof value !== 'object' || value._isProxy) {
-            return value;
-          }
+        let value = Reflect.get(target, property);
 
+        // Check if we need to add a proxy to this object
+        if (!!value && typeof value === 'object' && !value._isProxy && value._isProxyInternal) {
+          const isLeaf = !value._continueProxy;
           // Adding proxies to objects
-          // Intentionally only going 1 level deep
-          const handler = proxyHandlerLeaf(socket, property);
-          target[property] = new Proxy(value, handler);
-          target[property]._isProxy = true;
-
-          return target[property];
+          const handler = proxyHandlerChild(socket, property, isLeaf);
+          const proxiedValue = new Proxy(value, handler);
+          Reflect.set(target, property, proxiedValue);
+          value = proxiedValue;
         }
 
-        // This property is not in the context object yet
-        sendContextRequest(socket, property);
-        return null;
+        return value;
       }
     };
   }
 
-  // This handler is for leaf nodes only
-  const proxyHandlerLeaf = (socket, path) => {
+  const proxyHandlerChild = (socket, path, isLeaf) => {
     return {
       get: (target, property) => {
-        if (target[property] !== undefined) { // Only checking undefined properites for leaves
-          return target[property];
+        // Check if a proxy was already added to this object
+        if (property === '_isProxy') {
+          return true;
+        }
+
+        const value = Reflect.get(target, property);
+        const fullPropertyPath = `${path}.${property}`;
+
+        // If the value is undefined, we need to request it from the server
+        if (value === undefined && isLeaf) {
+          sendContextRequest(socket, fullPropertyPath);
```

Comment:
So for access into a nested object, we send a request at each level, right?

Does that mean that `matchupsInLeagueMap[leagueId][leg][rosterId] would trigger the following queries?

```
getMatchupsInLeague(leagueId)
getMatchupsInLeague(leagueId, leg)
getMatchupsInLeague(leagueId, leg, rosterId)
```

Reply:
So I was trying to avoid that, that's what the `isLeaf` value is supposed to check. `getMatchupsInLeague(leagueId, leg)` is the only one that should be done. The external api only requires 2 arguments.

Reply:
Ooh, got it. Missed the `isLeaf` part.